### ISSUE TITLE
Spring Boot 2.0.0.M3 project using spring-kafka requires spring-tx

### ISSUE THREAD

**Issue by luchob** [spring-projects/spring-kafka#406]
Hi, guys! 

This one is very similar to [384](https://github.com/spring-projects/spring-kafka/issues/384) but this time `spring-tx` is required as an additional dependency. If not provided the following exception is thrown at runtime:

```
2017-09-04 18:01:31.933 ERROR 13187 --- [           main] o.s.boot.SpringApplication               : Application startup failed

java.lang.NoClassDefFoundError: org/springframework/transaction/support/TransactionSynchronizationManager
	at org.springframework.kafka.core.KafkaTemplate.inTransaction(KafkaTemplate.java:333) ~[spring-kafka-2.0.0.M3.jar:na]
	at org.springframework.kafka.core.KafkaTemplate.doSend(KafkaTemplate.java:290) ~[spring-kafka-2.0.0.M3.jar:na]
	at org.springframework.kafka.core.KafkaTemplate.send(KafkaTemplate.java:170) ~[spring-kafka-2.0.0.M3.jar:na]
```
Explicitly adding `org.springframework:spring-tx` solves the issue. The reference to `spring-tx` is [here](https://github.com/spring-projects/spring-kafka/blob/7a42198ff51d751c4c45413d483b70e11e733ae9/spring-kafka/src/main/java/org/springframework/kafka/core/KafkaTemplate.java#L41).

I'm adding a simple `build.gradle` for reference (details are skipped for brevity):

```
buildscript {
	ext {
		springBootVersion = '2.0.0.M3'
	}
	...
	dependencies {
		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
	}
}

..
apply plugin: 'org.springframework.boot'
apply plugin: 'io.spring.dependency-management'
..

dependencies {
	compile('org.springframework.kafka:spring-kafka')
  //https://github.com/spring-projects/spring-kafka/issues/384
	compile('org.springframework.retry:spring-retry')
	compile('org.springframework.boot:spring-boot-starter-web')
}
```

**Comment by artembilan** [comment]
I think that's good catch and the code:
```
protected boolean inTransaction() {
		return this.producers.get() != null || TransactionSynchronizationManager.isActualTransactionActive();
}
```
Should be supplied with the check if `TransactionSynchronizationManager` is present in the classpath. Well actually even before calling this method.

**Comment by garyrussell** [comment]
Fixed on 1.3.x - needs to be forward ported to master.

**Comment by garyrussell** [comment]
https://github.com/spring-projects/spring-kafka/commit/316967d919f1dfd26875ff05479561c501944a18

@artembilan it can't be optional because there's a hard requirement in the container factory.
